### ISSUE TITLE
Add back navigation for most layouts to mitigate devices which lack a nav bar

### ISSUE THREAD

**Issue by BenHenning** [oppia/oppia-android#1128]
### Is your feature request related to a problem? Please describe.

On many modern Android devices, the **system navigation bar is hidden** (gesture navigation). This leads to accessibility issues in the Oppia app since users may not have a reliable way to navigate back from certain screens (e.g., Exploration Player, Question Player). To ensure a **consistent and accessible experience**, we should add a **back navigation button** in the toolbar for all eligible screens.

### Describe the solution you'd like

* Add a **toolbar back button** (`?attr/homeAsUpIndicator`) to all non-root screens that support back navigation.
* Provide a **content description** (`@string/navigate_up`) for accessibility.
* Hook the navigation to the standard back dispatcher:

```kotlin
toolbar.setNavigationOnClickListener {
  activity.finish()
}
```

This ensures that all users, regardless of navigation mode (gesture vs. navigation bar), can consistently navigate backward.
There should not be any anomalies between navigating back by gesture or navigation bar.
("By Gesture" means doing it physically by sliding the screen.)

### Scope

**Screens where back navigation should be added:**

* Exploration & Question flows
* Story/Topic Info Screens
* Profile Settings
* Admin Settings
* Any other nested detail screens

**Screens where back navigation should not be added:**

* Homescreen (exit instead of back navigation)
* Root-level tabs (e.g., Library, Profile tab)

### Implementation details

* Update XML layouts to include:

  ```xml
  <androidx.appcompat.widget.Toolbar
      android:id="@+id/toolbar"
      ...
      app:navigationContentDescription="@string/navigate_up"
      app:navigationIcon="?attr/homeAsUpIndicator" />
  ```

* In Fragment/Activity:

  ```kotlin
  toolbar.setNavigationOnClickListener {
    activity.finish()
  }
  ```

### Checklist

* [ ] **PR 1**: Add back navigation to Exploration Player, Question Player, and related flow screens.
* [ ] **PR 2**: Add back navigation to Story/Topic Info Screens.
* [ ] **PR 3**: Add back navigation to Profile Settings, Profile Adding, and Admin Settings screens.
* [ ] **PR 4**: Add back navigation to Remaining nested/detail screens.

### PR Submission Directions

* Submit PRs in chunks (see checklist above).
* Each PR should include:

  * Implementation of toolbar back button.
  * Screenshots and/or screen recordings showing the back navigation working.
  * Confirmation that TalkBack announces the **navigate up** button.
* Thoroughly test with:

  * Navigation bar visible
  * Gesture navigation mode (nav bar hidden)
  * Both portrait and landscape orientations

**Comment by BenHenning** [comment]
@mschanteltc do you have any thoughts on this?

**Comment by BenHenning** [comment]
@nikitamarysolomanpvt as well in case you have any thoughts.

**Comment by mschanteltc** [comment]
> @mschanteltc do you have any thoughts on this?

Correct me if I'm wrong, but I thought all Android apps have a fixed navbar unless specified otherwise? I think only iOS apps use back navigation in their headers, but since we're developing for Android, we can deal with this another time. Most apps have different menu functionalities between their iOS and Android applications, so maybe new mocks would be needed for iOS?

For Android though, are there any apps or devices that emulate this behavior?

**Comment by rt4914** [comment]
> > @mschanteltc do you have any thoughts on this?
> 
> Correct me if I'm wrong, but I thought all Android apps have a fixed navbar unless specified otherwise? I think only iOS apps use back navigation in their headers, but since we're developing for Android, we can deal with this another time. Most apps have different menu functionalities between their iOS and Android applications, so maybe new mocks would be needed for iOS?
> 
> For Android though, are there any apps or devices that emulate this behavior?

@mschanteltc As per my experience all major applications like Google Pay, Youtube, Instagram, Whatsapp uses back button navigation in android for internal screens (except home screen).
There is one exception to this which is Amazon-Shopping-App, which keeps same toolbar across all screens which contains logo, navigation drawer icon and search icon.

**Comment by mschanteltc** [comment]
You are right, having a back navigation that brings users to the previous page should be a function based on the material design [guidelines](https://material.io/design/navigation/understanding-navigation.html#reverse-navigation). I think the main pages that would need changing include the players and the topic pages. How does this sound?
